src/main/java/org/squiddev/plethora/integration/astralsorcery/IntegrationAstralSorcery.java

```diff
@@ -0,0 +1,777 @@
+			if (properties == null) return Collections.emptyMap();
+
+			Map<String, Object> out = new HashMap<>(4);
+
+			out.put("cutting", properties.getCollectiveCapability());
+			out.put("size", properties.getSize());
+			out.put("purity", properties.getPurity());
+
+			out.put("fracture", properties.getFracturation());
+
+			//This field doesn't appear to be used
+			//out.put("sizeOverride", properties.getSizeOverride());
+
+			return Collections.singletonMap("astralCrystal2", out);
+		}
+
+		@Nonnull
+		@Override
+		public ItemStack getExample() {
+			return ItemRockCrystalBase.createMaxBaseCrystal();
+		}
+	};
+*/
+
+	public static final IMetaProvider<CrystalProperties> META_CRYSTAL_PROPERTY = new BasicMetaProvider<CrystalProperties>(
+		"Provides the cutting, size, purity, and fracturing from this CrystalProperties"
+	) {
+		@Nonnull
+		@Override
+		public Map<String, Object> getMeta(@Nonnull CrystalProperties props) {
+			Map<String, Object> out = new HashMap<>(4);
+
+			out.put("cutting", props.getCollectiveCapability());
+			out.put("size", props.getSize());
+			out.put("purity", props.getPurity());
+
+			out.put("fracture", props.getFracturation());
+
+			//This field doesn't appear to be used
+			//out.put("sizeOverride", properties.getSizeOverride());
+
+			return Collections.singletonMap("crystalProperties", out);
+		}
+
+		@Nullable //If it returns Null, Astral broke something...
+		@Override
+		public CrystalProperties getExample() {
+			return CrystalProperties.getCrystalProperties(ItemRockCrystalBase.createMaxBaseCrystal());
+		}
+	};
+
+	public static final IMetaProvider<ItemStack> META_TUNED_CRYSTAL_BASE = new ItemStackContextMetaProvider<ItemTunedCrystalBase>(
+		ItemTunedCrystalBase.class,
+		"Provides the constellation(s) for this item"
+	) {
+		@Nonnull
+		@Override
+		public Map<String, Object> getMeta(@Nonnull IPartialContext<ItemStack> context, @Nonnull ItemTunedCrystalBase item) {
+			ItemStack stack = context.getTarget();
+			Map<String, Object> out = new HashMap<>(2);
+
+			IConstellation constellation = item.getFocusConstellation(stack);
+			out.put("constellationFocus", constellation == null ? null : context.makePartialChild(constellation).getMeta());
+
+			IConstellation trait = ItemTunedCrystalBase.getTrait(stack);
+			out.put("constellationTrait", trait == null ? null : context.makePartialChild(trait).getMeta());
+
+			return out;
+		}
+
+		@Nullable
+		@Override
+		public ItemStack getExample() {
+			ItemStack crystalStack = new ItemStack(ItemsAS.tunedRockCrystal);
+
+			IWeakConstellation constellation = getExampleConstellation();
+			if (constellation == null) return null;
+
+			ItemTunedCrystalBase.applyMainConstellation(crystalStack, constellation);
+
+			return crystalStack;
+		}
+	};
+
+	//Mainly useful if a player wants to design some sort of organizer to track their progress
+	public static final IMetaProvider<EntityPlayer> META_PLAYER_PROGRESS = new BaseMetaProvider<EntityPlayer>(
```

Comment:
I think we need to have a bit of a think about how this is exposed. It feels a little bit odd to just bundle it on the player entity, as it means you can get the progress for any player within a sensors range\*. I'd almost be inclined to add a method to the introspection module or something instead, but worth discussing first.

\* Though that would allow for some pretty cool security system, where only people with specific research can enter.

Reply:
I went ahead and moved this to a method, though it needs tested.